sdk/core/core-util/test/public/datetime.spec.ts

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { toOffsetDateTime } from "../../src/datetime";
+import { assert } from "chai";
+
+describe("utcDateTime with timezone offset", function () {
+  describe("work in +08:00 timezone", function () {
+    it("the timezone offset should be correct", function () {
+      const date = new Date("2022-01-05T19:08:10.000Z");
+      assert.strictEqual(
+        Date.parse(toOffsetDateTime(date)) - Date.parse(date.toISOString()),
+        60 * date.getTimezoneOffset() * 1000
```

Comment:
A little unsure whether the difference should be 0 ? 
 if it's `2022-01-05T19:08:10.000Z` In UTC time, then it should be `2022-01-06T03:08:10.000+08:00` in the +08:00 timezone. so maybe purely use something like the below in toOffsetDateTime is incorrect? 
`date.toISOString().replace(/Z$/, "") + diff + pad(tzOffset / 60) + ":" + pad(tzOffset % 60)`

Reply:
I think the difference should be 0 here, no matter the ISO datetime or offset datetime they are just format differences and the underlying number should be the one.

Not sure it's easy to convert them. If not, we may take the `offsetDateTime` as `utcDatetime` in [typespec](https://microsoft.github.io/typespec/next/standard-library/http/encoding#utcdatetime-and-offsetdatetime).

I notice that we didn't import any library to handle date and time. Not sure this is mainly for performance consideration. 

I know with momentjs we could easily format any date ([docs](http://momentjs.com/docs/#/displaying/format/))

```
> moment.defaultFormat
'YYYY-MM-DDTHH:mm:ssZ'
> moment.defaultFormatUtc
'YYYY-MM-DDTHH:mm:ss[Z]'
```

![image](https://github.com/Azure/azure-sdk-for-js/assets/9943211/a1554a36-ced2-46b3-b6df-f8e0cc2ef27f)

Reply:
My understanding is that moment is an extremely heavy dependency and something that will eventually be replaced by an ecmascript standard /cc @bterlson